cpp/include/cudf/utilities/error.hpp

```diff
@@ -46,7 +46,25 @@ struct logic_error : public std::logic_error {
  * @brief Exception thrown when a CUDA error is encountered.
  */
 struct cuda_error : public std::runtime_error {
-  cuda_error(std::string const& message) : std::runtime_error(message) {}
+  cuda_error(std::string const& message, cudaError_t const& error)
+    : std::runtime_error(message), _cudaError(error)
+  {
+  }
+  cudaError_t error_code() { return _cudaError; }
+
+ protected:
+  cudaError_t _cudaError;
+};
+
+struct cudart_error : public cuda_error {
+  cudart_error(std::string const& message, cudaError_t const& error) : cuda_error(message, error) {}
+};
+
+struct sticky_cuda_error : public cuda_error {
```

Comment:
I think `fatal_cuda_error` is still a more appropriate name.

Reply:
Changed.